pandas/_libs/tslibs/tzconversion.pyx

```diff
@@ -416,8 +416,11 @@ timedelta-like}
 
                 else:
                     delta_idx = bisect_right_i8(info.tdata, new_local, info.ntrans)
-
-                    delta_idx = delta_idx - delta_idx_offset
+                    if (shift_forward or shift_delta > 0) and \
+                       info.deltas[delta_idx-1] >= 0:
+                        delta_idx = delta_idx - 1
```

Comment:
Could you write a 1 line comment describing why we need to do this?

Reply:
added